docs/source/checks/vision/model_evaluation/plot_weak_segments_performance.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+"""
+Weak Segments Performance
+*************************
+
+This notebook provides an overview for using and understanding the weak segment performance check.
+
+**Structure:**
+
+* `What is the purpose of the check? <#what-is-the-purpose-of-the-check>`__
+* `Automatically detecting weak segments <#automatically-detecting-weak-segments>`__
+* `Generate data & model <#generate-data-model>`__
+* `Run the check <#run-the-check>`__
+* `Define a condition <#define-a-condition>`__
+
+What is the purpose of the check?
+==================================
+
+The check is designed to help you easily identify the model's weakest segments in the data provided.
+The segments are characterized by the image properties.
+
+Automatically detecting weak segments
+=====================================
+
+The check contains several steps:
+
+#. We calculate the image properties for each sample. The properties to calculate can be passed explicitly or set to the
+   default image properties.
+
+#. We calculate loss for each sample in the dataset using the provided model, the loss function can be passed explicitly
+   or set to a default by the task type.
+
+#. We train multiple simple tree based models, each one is trained using exactly two
+   properties to predict the per sample error calculated before.
+
+#. We convert each of the leafs in each of the trees into a segment and calculate the segment's performance. For the
+   weakest segments detected we also calculate the model's performance on data segments surrounding them.
```

Comment:
```suggestion
   weakest segments detected we also calculate the model's performance on the data segments surrounding them.
```
Also should probably better define "surrounding"